### DISCUSSION TITLE
error: failed to push some refs

### DISCUSSION THREAD

**Discussion by caosuyang** [git-lfs/git-lfs#5835] [Q&A]
git --no-optional-locks -c color.branch=false -c color.diff=false -c color.status=false -c diff.mnemonicprefix=false -c core.quotepath=false -c credential.helper=sourcetree push -v origin refs/heads/feature/ryancao/build-errors:refs/heads/feature/ryancao/build-errors 
Pushing to github.com:caosuyang/mzzs-bwk2020.git
remote: error: Trace: 92695125cf1cbdd88462212fa828d90ae78ca074e35c661864209f9c7be190a7        
remote: error: See https://gh.io/lfs for more information.        
remote: error: File bwk2020/BWKOEM/Legacy/Template/Component/TXLiteAVSDK_Player/SDK/TXLiteAVSDK_Player.framework/TXLiteAVSDK_Player is 136.11 MB; this exceeds GitHub's file size limit of 100.00 MB        
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.        
To github.com:caosuyang/mzzs-bwk2020.git
 ! [remote rejected] feature/ryancao/build-errors -> feature/ryancao/build-errors (pre-receive hook declined)
error: failed to push some refs to 'github.com:caosuyang/mzzs-bwk2020.git'
Completed with errors, see above

➜  mzzs-bwk2020 git:(feature/ryancao/build-errors) ✗ git lfs track "bwk2020/BWKOEM/Legacy/CurrentApp/Base/Framework/Libs/libPlayerSDKLib.a"
Tracking "bwk2020/BWKOEM/Legacy/CurrentApp/Base/Framework/Libs/libPlayerSDKLib.a"
➜  mzzs-bwk2020 git:(feature/ryancao/build-errors) ✗ git lfs track "bwk2020/BWKOEM/Legacy/Template/Component/TXLiteAVSDK_Player/SDK/TXLiteAVSDK_Player.framework/TXLiteAVSDK_Player“
dquote>
➜  mzzs-bwk2020 git:(feature/ryancao/build-errors) ✗ git lfs track "bwk2020/BWKOEM/Legacy/Template/Component/TXLiteAVSDK_Player/SDK/TXLiteAVSDK_Player.framework/TXLiteAVSDK_Player"
Tracking "bwk2020/BWKOEM/Legacy/Template/Component/TXLiteAVSDK_Player/SDK/TXLiteAVSDK_Player.framework/TXLiteAVSDK_Player"

bwk2020/BWKOEM/Template/Component/TXLiteAVSDK_Player/SDK/TXLiteAVSDK_Player.framework/TXLiteAVSDK_Player filter=lfs diff=lfs merge=lfs -text
bwk2020/BWKOEM/CurrentApp/Base/Framework/Libs/libPlayerSDKLib.a filter=lfs diff=lfs merge=lfs -text
bwk2020/BWKOEM/Legacy/Template/Component/TXLiteAVSDK_Player/SDK/TXLiteAVSDK_Player.framework/TXLiteAVSDK_Player filter=lfs diff=lfs merge=lfs -text
bwk2020/BWKOEM/Legacy/CurrentApp/Base/Framework/Libs/libPlayerSDKLib.a filter=lfs diff=lfs merge=lfs -text

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.  My guess here is that you have a file that's already in your Git history, and so just running `git lfs track` will not suffice to convert your pre-existing Git blobs into Git LFS objects.  For that, you need to use the `git lfs migrate import` command to migrate your Git history to use Git LFS.

I've recently written up some detailed responses to similar questions in https://github.com/git-lfs/git-lfs/discussions/5752#discussioncomment-9484852 and https://github.com/git-lfs/git-lfs/discussions/5817#discussioncomment-9939137; you might take a look at those for further advice.

In brief, you'll likely want to run the following and then you'll need to `git push --force` to overwrite your remote's Git history with the new one.  But please do be careful to ensure you understand the consequences of those steps; you are changing your Git history with them, so be cautious and make sure you are happy with the changes.

And again, reading through the two responses I mentioned should provide more help and context on these steps.

```
$ git lfs migrate import --everything \
  --include "bwk2020/BWKOEM/CurrentApp/Base/Framework/Libs/libPlayerSDKLib.a" \
  --include "bwk2020/BWKOEM/Legacy/CurrentApp/Base/Framework/Libs/libPlayerSDKLib.a" \
  --include "bwk2020/BWKOEM/Legacy/Template/Component/TXLiteAVSDK_Player/SDK/TXLiteAVSDK_Player.framework/TXLiteAVSDK_Player" \
  --include "bwk2020/BWKOEM/Template/Component/TXLiteAVSDK_Player/SDK/TXLiteAVSDK_Player.framework/TXLiteAVSDK_Player"
```
